usage/rest-server/src/main/java/brooklyn/rest/resources/CatalogResource.java

```diff
@@ -311,6 +317,18 @@ public Response getIcon(String itemId, String version) {
         return getCatalogItemIcon(result);
     }
 
+    @Override
+    public void setDeprecated(String itemId, boolean deprecated) {
+        if (!Entitlements.isEntitled(mgmt().getEntitlementManager(), Entitlements.MODIFY_CATALOG_ITEM, StringAndArgument.of(itemId, "deprecated"))) {
+            throw WebResourceUtils.unauthorized("User '%s' is not authorized to modify catalog",
+                    Entitlements.getEntitlementContext().user());
+        }
+        CatalogItem<?, ?> item = CatalogUtils.getCatalogItemOptionalVersion(mgmt(), itemId);
+        if (item==null)
+            throw WebResourceUtils.notFound("Catalog item with id '%s' not found", itemId);
+        item.setDeprecated(deprecated);
```

Comment:
You presumably need to call `catalog.persist(item)` after doing the `item.setDeprecated`?